src/backend/utils/mmgr/mcxt.c

```diff
@@ -794,13 +823,25 @@ MemoryContextStats_recur(MemoryContext topContext, MemoryContext rootContext,
 			{
 				if (siblingCount != 0)
 				{
+#ifdef EXTRA_DYNAMIC_MEMORY_DEBUG
+					/*
+					   Save current chunk_htable to temp_chunk_htab, restore
+					   prev_chunk_htable to chunk_htable, print logs and restore
+					   current chunk_htable from temp_chunk_htab.
+					 */
+					HTAB * temp_chunks_htable = chunks_htable;
```

Comment:
Please take a look to this part and try to propose a solution which avoids code duplication.